packages/cli/src/backend-identifier/backend_identifier_with_sandbox_fallback.ts

```diff
@@ -18,23 +18,25 @@ export class BackendIdentifierResolverWithFallback
     private fallbackResolver: SandboxBackendIdResolver
   ) {}
   /**
-   * resolves the backend id, falling back to the sandbox id if there is an error
+   * resolves to deployed backend id, falling back to the sandbox id if there is an error and stack, appId, and branch are not provided
    */
   resolveDeployedBackendIdentifier = async (
     args: BackendIdentifierParameters
   ) => {
-    return (
-      (await this.defaultResolver.resolveDeployedBackendIdentifier(args)) ??
-      (await this.fallbackResolver.resolve())
-    );
+    if (args.stack || args.appId || args.branch) {
+      return await this.defaultResolver.resolveDeployedBackendIdentifier(args);
```

Comment:
no need to await if we are not doing error handling.

Reply:
We could possibly be throwing an error in the namespace resolver.

Reply:
that will be caught by the caller when they `await` this method.